message-publisher/src/main/java/io/eventuate/examples/tram/vanilla/event/publishing/JdbcStatementExecutor.java

```diff
@@ -0,0 +1,52 @@
+package io.eventuate.examples.tram.vanilla.event.publishing;
+
+import com.zaxxer.hikari.HikariDataSource;
+import io.eventuate.common.jdbc.EventuateJdbcStatementExecutor;
+import io.eventuate.common.jdbc.EventuateRowMapper;
+import io.eventuate.common.jdbc.EventuateSqlException;
+import org.apache.commons.lang.NotImplementedException;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+
+public class JdbcStatementExecutor implements EventuateJdbcStatementExecutor {
+
+  private HikariDataSource dataSource;
+
+  public JdbcStatementExecutor(String user, String password, String driver, String url) {
```

Comment:
I think that `JdbcStatementExecutor` should be injected with 

```
interface JdbcConnectionSupplier {
  void doWithConnection(Consumer<Connection> connectionConsumer);
}
```

All the code such as `new HikariDataSource();` and `try(Connection connection = dataSource.getConnection();` should be moved into a concrete implementation of `JdbcConnectionSupplier`.

In a legacy application `JdbcConnectionSupplier` would need to use the connection provided by the legacy application.

Reply:
ok